### PR TITLE
Reorder the checkboxes in the PR template

### PR DESCRIPTION
**Pull request by Hixie**
...so that they are listed in more or less the likely order that people will do them in.

## Pre-launch Checklist

- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (stuartmorgan) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: New contributors generally do this after creating the PR since that's when the bot tells them to. I'm not sure whether the ordering is intended to be new or established contributors; I would have thought the former since for established contributors the checklist is much more rote.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (stuartmorgan) [inline .github/PULL_REQUEST_TEMPLATE.md:16]: I would probably put these two last.

**Review by stuartmorgan — APPROVED** [review]
LGTM. Some optional thoughts.

(I'm curious whether you've found the checklist to actually be effective as a checklist; my experience has been that people mostly just check all the boxes whether they've done them or not. E.g., my impression from triage in this repo is that most PRs that don't list an issue have the box checked for listing an issue.)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Hixie) [reply .github/PULL_REQUEST_TEMPLATE.md:12]: Well to be frank I put the order in the order that I usually complete them, but in any case, wouldn't the order be the same here for new or established contributors? New ones should hopefully read the contributor guide first, then the other two, before they write their PR, and certainly before they get it reviewed, but they don't need to sign the CLA until it's ready to land, really. And established contributors have already done all four.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Hixie) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: Well until you update the changelog, the tests can't pass, so I figured the tests one had to go last, and then for thematic consistency I put the other test one with it...

**Comment by Hixie** [comment]
> (I'm curious whether you've found the checklist to actually be effective as a checklist; my experience has been that people mostly just check all the boxes whether they've done them or not. E.g., my impression from triage in this repo is that most PRs that don't list an issue have the box checked for listing an issue.)

I would take that as a sign that the person isn't reading our contributing materials and treat the PR accordingly (e.g. close it with a polite comment saying that it seems the instructions were not followed).

I originally found these templates annoying but after the recent edits to make them more streamlined I've ended up finding I do use them, they're a useful way to make sure I remembered everything especially in repos I'm not frequently contributing to.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (stuartmorgan) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: There's only a failure from not updating the CHANGELOG of you update the version, so as long as these two stay together it won't affect tests.

Comment (Hixie) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: Surely the checkbox for "all the tests pass" has to go last though? Since as soon as you make any change the tests have to run again, so you can't know if the tests have passed yet? Seems like by definition you have to check the "changelog" one before the "tests" one...

Anyway I don't feel strongly about this at all, I'm happy to defer to whatever order you'd like to see. Let me know what order you want it in and I'll land that. :-)

Comment (stuartmorgan) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: I guess that makes sense. LGTM as-is.

**Review by stuartmorgan — APPROVED** [review]

**Pull request merged**
